### PR TITLE
Jvojtek/module refactor

### PR DESCRIPTION
**Pull request by JaroVojtek**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -48,39 +48,80 @@ variable "helm_create_namespace" {
...
+variable "irsa_policy_enabled" {
+  type        = bool
+  default     = true
+  description = "Whether to create opinionated policy to allow operations on specified zones in `policy_allowed_zone_ids`."
+}
+
+variable "irsa_assume_role_enabled" {
+  type        = bool
+  default     = false
+  description = "Whether IRSA is allowed to assume role defined by assume_role_arn. Useful for hosted zones in another AWS account."
```
Comment (dojci) [inline variables.tf:84]: fix description typo, we are not using hosted zones in this module

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -9,29 +9,32 @@ repos:
...
       - id: detect-secrets
         args: ['--baseline', '.secrets.baseline']
+        exclude: terraform.tfstate
+
+  - repo: https://github.com/alessandrojcm/commitlint-pre-commit-hook
+    rev: v8.0.0
+    hooks:
+        - id: commitlint
+          stages: [commit-msg]
+          additional_dependencies: ['@commitlint/config-conventional']
```
Comment (dojci) [inline .pre-commit-config.yaml:40]: please remove commit lint validation

File: variables.tf (hunk 1 above)
Comment (JaroVojtek) [reply variables.tf]: fixed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (JaroVojtek) [reply .pre-commit-config.yaml]: removed

**Comment by JaroVojtek** [comment]
tests added in basic/examples

File: commitlint.config.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module.exports = {
```
Comment (dojci) [inline commitlint.config.js:1]: please remove unnecessary file after we removed commit linting from pre commit hooks

**Review by dojci — CHANGES_REQUESTED** [review]

**Review by dojci — APPROVED** [review]

**Pull request merged**